src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -520,6 +523,13 @@ ColumnsDescription InterpreterCreateQuery::getColumnsDescription(
         /// add column to postprocessing if there is a default_expression specified
         if (col_decl.default_expression)
         {
+            // substitute possible UDFs with their definitions
+            if (!UserDefinedSQLFunctionFactory::instance().empty())
```

Comment:
This solution is good, but we have our AST implementation problem. If you check ASTColumnDeclaration you can see that there are a bunch of members (`default_expression`), and there are also children array and they are not synchronized. If we only change member, children array will not be updated and it can become a big issue, with very hard to debug bugs.
Last time when I tried to do inline UDF in table definition, I tried to replace `ORDER BY`, `PARTITION BY` definitions in table, then there were a bunch of crashes, because `ASTStorage` does not keep shared members, but just raw pointers.
In QueryTree, there are no public members, only public getters, and setters that operate on children array. The only solution without redesigning our AST children, to make it work, it seems to just write visitor that is aware of our AST implementation problems, and in this visitor we need special handling for `ASTColumnDeclaration`, `ASTStorage`.

Reply:
Yes, I noticed that issue, not in form of crash but with inconsistencies. If you modify the create AST with the visitor, the tree will be correct but the `default_expression` will contain the old definition which at later steps will be used making the visitor changes obsolete. Really annoying to debug.

I'll try then with modifying the visitor 👍

Reply:
Would this be what you had in mind?
I can expand with extra checks, wanted to be sure about the idea first.

Also, does it make sense for me to try to expand with `ORDER BY`, `PARTITION BY` inlining? Because the same problem should occur as in the linked issues if UDFs are used there.